### ISSUE TITLE
Generated models do not implement user defined interfaces

### ISSUE THREAD

**Issue by samlitowitz** [99designs/gqlgen#716]
### What happened?
Generated models do not fulfill user defined interfaces provided through models configuration.
```go
// models/generated.go
// Code generated by github.com/99designs/gqlgen, DO NOT EDIT.

package models

type MyObect struct {
	ID string `json:"id"`
}

func (MyObect) IsMyInterface() {}
```

### What did you expect?
An error preventing me from doing this, OR the generated models need to fulfill the user defined interface, i.e.
```go
// models/generated.go
// Code generated by github.com/99designs/gqlgen, DO NOT EDIT.

package models

type MyObect struct {
	ID string `json:"id"`
}

func (MyObect) Baz() {}
```

### Minimal graphql.schema and models to reproduce
```yml
# .gqlgen.yml
exec:
  filename: generated/exec.go
model:
  filename: models/generated.go

models:
  MyInterface:
    model: github.com/99designs/gqlgen/example/interface_failure/models.MyInterface

```
```graphql
# schema.graphql
interface MyInterface  {
    id: ID!
}

type MyObect implements MyInterface {
    id: ID!
}

type Query {
    myObjects: MyObect!
}
```
```go
// models/my_interface.go
package models

type MyInterface interface {
	Baz()
}
```

### versions
 - `gqlgen version`? `dev`,  from master with test case added to the `example/` directory
 - `go version`? `go version go1.12.4 linux/amd64`
 - dep or go modules? master branch [go.mod](https://github.com/99designs/gqlgen/blob/master/go.mod)

**Comment by hh** [comment]
I also found this working through https://gqlgen.com/getting-started/

See https://github.com/ii/org/blob/master/research/gqlgen.org#errors

Where I was writing up an org file for it and ran into:

```
-*- mode: compilation; default-directory: "~/gqlgen-todos/" -*-
Compilation started at Sun May 19 07:11:45

go run server/server.go 
# github.com/hh/gqlgen-todos
./resolver.go:17:24: cannot use &queryResolver literal (type *queryResolver) as type QueryResolver in return argument:
	*queryResolver does not implement QueryResolver (wrong type for Todos method)
		have Todos(context.Context) ([]Todo, error)
		want Todos(context.Context) ([]*Todo, error)

Compilation exited abnormally with code 2 at Sun May 19 07:11:45
```

**Comment by hh** [comment]
I added my environment:

https://github.com/ii/org/blob/master/research/gqlgen.org#my-environment

  - [environ](#sec-1)
  - [go env](#sec-2)

# environ<a id="sec-1"></a>

```shell
go version
```

    go version go1.12.3 linux/amd64

# go env<a id="sec-2"></a>

```shell
go env
```

    GOARCH="amd64"
    GOBIN=""
    GOCACHE="/home/hippie/.cache/go-build"
    GOEXE=""
    GOFLAGS=""
    GOHOSTARCH="amd64"
    GOHOSTOS="linux"
    GOOS="linux"
    GOPATH="/home/hippie/go"
    GOPROXY=""
    GORACE=""
    GOROOT="/usr/local/go"
    GOTMPDIR=""
    GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
    GCCGO="gccgo"
    CC="gcc"
    CXX="g++"
    CGO_ENABLED="1"
    GOMOD="/home/hippie/gqlgen-todos/go.mod"
    CGO_CFLAGS="-g -O2"
    CGO_CPPFLAGS=""
    CGO_CXXFLAGS="-g -O2"
    CGO_FFLAGS="-g -O2"
    CGO_LDFLAGS="-g -O2"
    PKG_CONFIG="pkg-config"
    GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build045453696=/tmp/go-build -gno-record-gcc-switches"

**Comment by hh** [comment]
I updated the documentation to fix #714  in #718.
